# NRC Emotional Lexicon

This is the [NRC Emotional Lexicon](http://saifmohammad.com/WebPages/NRC-Emotion-Lexicon.htm): "The NRC Emotion Lexicon is a list of English words and their associations with eight basic emotions (anger, fear, anticipation, trust, surprise, sadness, joy, and disgust) and two sentiments (negative and positive). The annotations were manually done by crowdsourcing."

http://saifmohammad.com/WebPages/NRC-Emotion-Lexicon.htm

In [65]:
import pandas as pd
from nrclex import NRCLex

In [17]:
filepath = r"C:\Users\nitis\OneDrive\Desktop\Innovation lab\dictionary.txt"
emolex_df = pd.read_csv(filepath,  names=["word", "emotion", "association"], skiprows=45, sep='\t', keep_default_na=False)
emolex_df.head(12)

word       emotion  association
0    aback         anger            0
1    aback  anticipation            0
2    aback       disgust            0
3    aback          fear            0
4    aback           joy            0
5    aback      negative            0
6    aback      positive            0
7    aback       sadness            0
8    aback      surprise            0
9    aback         trust            0
10  abacus         anger            0
11  abacus  anticipation            0

Seems kind of simple. A column for a word, a column for an emotion, and whether it't associated or not. You see "aback aback aback aback" because there's a row for every word-emotion pair.

## What emotions are covered?

Let's look at the 'emotion' column. What can we talk about?

In [18]:
emolex_df.emotion.unique()

array(['anger', 'anticipation', 'disgust', 'fear', 'joy', 'negative',
       'positive', 'sadness', 'surprise', 'trust'], dtype=object)

In [19]:
emolex_df.emotion.value_counts()

positive        14182
negative        14182
disgust         14182
anger           14182
anticipation    14182
joy             14182
surprise        14182
fear            14182
sadness         14182
trust           14182
Name: emotion, dtype: int64

## How many words does each emotion have?

Each emotion doesn't have 14182 words associated with it, unfortunately! `1` means "is associated" and `0` means "is not associated."

We're only going to care about "is associated."

In [20]:
emolex_df[emolex_df.association == 1].emotion.value_counts()

negative        3324
positive        2312
fear            1476
anger           1247
trust           1231
sadness         1191
disgust         1058
anticipation     839
joy              689
surprise         534
Name: emotion, dtype: int64

In theory things could be *kind of* angry or *kind of* joyous, but it doesn't work like that. If you want to spend a few hundred dollars on Mechnical Turk, though, *your own personal version can.*

## What if I just want the angry words?

In [21]:
emolex_df[(emolex_df.association == 1) & (emolex_df.emotion == 'anger')].word

30          abandoned
40        abandonment
170             abhor
180         abhorrent
270           abolish
             ...     
141220       wrongful
141230        wrongly
141470           yell
141500           yelp
141640          youth
Name: word, Length: 1247, dtype: object

## Reshaping

You can also reshape the data in order to look at it a slightly different way

In [22]:
emolex_words = emolex_df.pivot(index='word', columns='emotion', values='association').reset_index()
emolex_words.head()

emotion         word  anger  anticipation  disgust  fear  joy  negative  \
0              aback      0             0        0     0    0         0   
1             abacus      0             0        0     0    0         0   
2            abandon      0             0        0     1    0         1   
3          abandoned      1             0        0     1    0         1   
4        abandonment      1             0        0     1    0         1   

emotion  positive  sadness  surprise  trust  
0               0        0         0      0  
1               0        0         0      1  
2               0        1         0      0  
3               0        1         0      0  
4               0        1         1      0

You can now pull out individual words...

In [23]:
# If you didn't reset_index you could do this more easily
# by doing emolex_words.loc['charitable']
emolex_words[emolex_words.word == 'charitable']

emotion        word  anger  anticipation  disgust  fear  joy  negative  \
2001     charitable      0             1        0     0    1         0   

emotion  positive  sadness  surprise  trust  
2001            1        0         0      1

...or individual emotions....

In [24]:
emolex_words[emolex_words.anger == 1].head()

emotion         word  anger  anticipation  disgust  fear  joy  negative  \
3          abandoned      1             0        0     1    0         1   
4        abandonment      1             0        0     1    0         1   
17             abhor      1             0        1     1    0         1   
18         abhorrent      1             0        1     1    0         1   
27           abolish      1             0        0     0    0         1   

emotion  positive  sadness  surprise  trust  
3               0        1         0      0  
4               0        1         1      0  
17              0        0         0      0  
18              0        0         0      0  
27              0        0         0      0

...or multiple emotions!

In [25]:
emolex_words[(emolex_words.joy == 1) & (emolex_words.negative == 1)].head()

emotion        word  anger  anticipation  disgust  fear  joy  negative  \
61        abundance      0             1        1     0    1         1   
1018           balm      0             1        0     0    1         1   
1382     boisterous      1             1        0     0    1         1   
1916      celebrity      1             1        1     0    1         1   
2004        charmed      0             0        0     0    1         1   

emotion  positive  sadness  surprise  trust  
61              1        0         0      1  
1018            1        0         0      0  
1382            1        0         0      0  
1916            1        0         1      1  
2004            1        0         0      0

The useful part is going to be just getting words for a **single emotion.**

In [26]:
# Angry words
emolex_words[emolex_words.anger == 1].word

3          abandoned
4        abandonment
17             abhor
18         abhorrent
27           abolish
            ...     
14122       wrongful
14123        wrongly
14147           yell
14150           yelp
14164          youth
Name: word, Length: 1247, dtype: object

# NRC LExion pakage 

In [70]:
f = open(r"C:\Users\nitis\OneDrive\Desktop\Innovation lab\data1.txt")
text =(f.read()) 
text

'he Lake Waikaremoana Track is one of three Great Walks in New Zealand’s North Island. These Great Walks are hiking tracks, deemed to showcase some of the best natural landscapes in the country, with the trails and accommodation maintained to a far higher standard than your typical backcountry hut.\n\nLake Waikaremoana is one of the lesser-walked tracks, especially in comparison to the big hitters down in the South Island, which get booked out months in advance. And sure, Lake Waikaremoana doesn’t have those snow-capped mountains and pristine golden-sand beaches, but its scenery is impressive nonetheless. And the lack of crowds allow you to truly disconnect and immerse yourself in this spiritual destination.\n\nI loved walking the Lake Waikaremoana Track and highly recommend it as a multi-day adventure in New Zealand. Today, I’m going to be sharing absolutely everything you need to know about preparing for and tackling this fantastic walk.\n\nLet’s get started. '

In [73]:
#https://pypi.org/project/NRCLex/
text_object = NRCLex(text)

In [74]:
#return words list
text_object.words

['he',
 'Lake',
 'Waikaremoana',
 'Track',
 'is',
 'one',
 'of',
 'three',
 'Great',
 'Walks',
 'in',
 'New',
 'Zealand’s',
 'North',
 'Island',
 'These',
 'Great',
 'Walks',
 'are',
 'hiking',
 'tracks',
 'deemed',
 'to',
 'showcase',
 'some',
 'of',
 'the',
 'best',
 'natural',
 'landscapes',
 'in',
 'the',
 'country',
 'with',
 'the',
 'trails',
 'and',
 'accommodation',
 'maintained',
 'to',
 'a',
 'far',
 'higher',
 'standard',
 'than',
 'your',
 'typical',
 'backcountry',
 'hut',
 'Lake',
 'Waikaremoana',
 'is',
 'one',
 'of',
 'the',
 'lesser-walked',
 'tracks',
 'especially',
 'in',
 'comparison',
 'to',
 'the',
 'big',
 'hitters',
 'down',
 'in',
 'the',
 'South',
 'Island',
 'which',
 'get',
 'booked',
 'out',
 'months',
 'in',
 'advance',
 'And',
 'sure',
 'Lake',
 'Waikaremoana',
 'doesn’t',
 'have',
 'those',
 'snow-capped',
 'mountains',
 'and',
 'pristine',
 'golden-sand',
 'beaches',
 'but',
 'its',
 'scenery',
 'is',
 'impressive',
 'nonetheless',
 'And',
 'the',
 

In [75]:
#return sentences list
text_object.sentences

[Sentence("he Lake Waikaremoana Track is one of three Great Walks in New Zealand’s North Island."),
 Sentence("These Great Walks are hiking tracks, deemed to showcase some of the best natural landscapes in the country, with the trails and accommodation maintained to a far higher standard than your typical backcountry hut."),
 Sentence("Lake Waikaremoana is one of the lesser-walked tracks, especially in comparison to the big hitters down in the South Island, which get booked out months in advance."),
 Sentence("And sure, Lake Waikaremoana doesn’t have those snow-capped mountains and pristine golden-sand beaches, but its scenery is impressive nonetheless."),
 Sentence("And the lack of crowds allow you to truly disconnect and immerse yourself in this spiritual destination."),
 Sentence("I loved walking the Lake Waikaremoana Track and highly recommend it as a multi-day adventure in New Zealand."),
 Sentence("Today, I’m going to be sharing absolutely everything you need to know about 

In [76]:
#return affect list
text_object.affect_list

['positive',
 'positive',
 'sadness',
 'anticipation',
 'fear',
 'joy',
 'positive',
 'surprise',
 'positive',
 'negative',
 'negative',
 'sadness',
 'anticipation',
 'fear',
 'joy',
 'positive',
 'surprise',
 'trust',
 'anticipation',
 'fear',
 'joy',
 'positive',
 'sadness',
 'surprise',
 'positive',
 'trust',
 'anticipation',
 'positive']

In [77]:
# return affect dictionary
text_object.affect_dict

{'accommodation': ['positive'],
 'hut': ['positive', 'sadness'],
 'advance': ['anticipation', 'fear', 'joy', 'positive', 'surprise'],
 'pristine': ['positive'],
 'lack': ['negative'],
 'disconnect': ['negative', 'sadness'],
 'immerse': ['anticipation', 'fear', 'joy', 'positive', 'surprise', 'trust'],
 'destination': ['anticipation',
  'fear',
  'joy',
  'positive',
  'sadness',
  'surprise'],
 'recommend': ['positive', 'trust'],
 'adventure': ['anticipation', 'positive']}

In [78]:
##Return raw emotional counts
text_object.raw_emotion_scores


{'positive': 8,
 'sadness': 3,
 'anticipation': 4,
 'fear': 3,
 'joy': 3,
 'surprise': 3,
 'negative': 2,
 'trust': 2}

In [79]:
#return highest emotions
text_object.top_emotions

[('positive', 0.2857142857142857)]

In [80]:
#return affect frequescies
text_object.affect_frequencies

{'fear': 0.10714285714285714,
 'anger': 0.0,
 'anticip': 0.0,
 'trust': 0.07142857142857142,
 'surprise': 0.10714285714285714,
 'positive': 0.2857142857142857,
 'negative': 0.07142857142857142,
 'sadness': 0.10714285714285714,
 'disgust': 0.0,
 'joy': 0.10714285714285714,
 'anticipation': 0.14285714285714285}